### PR TITLE
apis: add core sched apis

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

koordlet: add the apis for the Linux [Core Scheduling](https://docs.kernel.org/admin-guide/hw-vuln/core-scheduling.html) feature.

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: apis/slo/v1alpha1/pod.go [hunk 1]
```diff
@@ -64,3 +64,19 @@ func GetPodMemoryQoSConfig(pod *corev1.Pod) (*PodMemoryQOSConfig, error) {
...
+	AnnotationPodEnableCoreSched = apiext.DomainPrefix + "enable-core-sched"
+
+	// AnnotationCoreSchedGroupID is the annotation key of the group ID of the Core Sched.
+	// Core Sched: https://docs.kernel.org/admin-guide/hw-vuln/core-scheduling.html
+	// Pods with the different core sched group IDs will not be running at the same SMT core at the same time, which
+	// means they will take different core sched cookies. And a pod with cookie is allowed to run on the same core with
+	// a pod who does not set any cookie (cookie=0).
+	// NOTE: If a pod sets the core sched enabled but misses the core sched ID, the pod will be assigned an individual
+	// and stateless internal ID according to the pod's UID.
+	AnnotationCoreSchedGroupID = apiext.NodeDomainPrefix + "/core-sched-group-id"
```
Comment (eahydra) [inline apis/slo/v1alpha1/pod.go:81]: Is there any format or specification requirement for the value corresponding to this annotation?

Comment (saintube) [reply apis/slo/v1alpha1/pod.go:83]: added

File: apis/slo/v1alpha1/pod.go [hunk 2]
```diff
@@ -64,3 +65,49 @@ func GetPodMemoryQoSConfig(pod *corev1.Pod) (*PodMemoryQOSConfig, error) {
...
+	// AnnotationCoreSchedGroupID is the annotation key of the group ID of the Core Sched.
+	// Value is expected to be a valid UUID.
+	// Core Sched: https://docs.kernel.org/admin-guide/hw-vuln/core-scheduling.html
+	// NOTE: Pods with the different core sched group IDs will not be running at the same SMT core at the same time,
+	//       which means they will take different core sched cookies. If a pod sets the core sched disabled, it will
+	//       take the default core sched cookie (0) and will also be force-idled to run on the same SMT core
+	//       concurrently with the core-sched-enabled pods.
+	//       If a pod sets the core sched enabled but misses the core sched ID, the pod will be assigned an individual
+	//       and stateless internal ID according to the pod's UID.
+	AnnotationCoreSchedGroupID = apiext.NodeDomainPrefix + "/core-sched-group-id"
```
Comment (hormes) [inline apis/slo/v1alpha1/pod.go:85]: Why not consider merging the above two fields? Setting core-sched-group-id means enabling core-sched.

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 3]
```diff
@@ -25,10 +25,21 @@ import (
 
 // CPUQOS enables cpu qos features.
 type CPUQOS struct {
+	// applied policy for the CPU QoS, default = "groupIdentity"
+	Policy *CPUQOSPolicy `json:"policy,omitempty"`
```
Comment (hormes) [inline apis/slo/v1alpha1/nodeslo_types.go:29]: Will these two strategies exist at the same time?

Comment (saintube) [reply apis/slo/v1alpha1/nodeslo_types.go:29]: No, they can not co-exist.

File: apis/slo/v1alpha1/pod.go (hunk 2 above)
Comment (saintube) [reply apis/slo/v1alpha1/pod.go:85]: The core scheduling can be enabled at node-level via NodeSLO where all pods are set cookies. So there may be a case that some pods disable the core sched (use the default cookie 0) while other pods enabled due to the NodeSLO spec.

Comment (eahydra) [inline apis/slo/v1alpha1/pod.go:85]: The annotation's prefix is NodeDomainPrefix, but only Pods can use the key, so is this a typo? Also I would recommend merging these two annotations.

Comment (saintube) [reply apis/slo/v1alpha1/pod.go]: merged into one annotation

**Review by jasonliu747 — APPROVED** [review]
/lgtm

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 4]
```diff
@@ -26,9 +26,27 @@ import (
 // CPUQOS enables cpu qos features.
 type CPUQOS struct {
 	// group identity value for pods, default = 0
+	// NOTE: It takes effect if CPUPolicy = "groupIdentity".
 	GroupIdentity *int64 `json:"groupIdentity,omitempty" validate:"omitempty,min=-1,max=2"`
+	// cpu.idle value for pods, default = 0.
+	// `1` means using SCHED_IDLE.
+	// CGroup Idle (introduced since mainline Linux 5.15): https://lore.kernel.org/lkml/162971078674.25758.15464079371945307825.tip-bot2@tip-bot2/#r
+	// NOTE: It takes effect if CPUPolicy = "coreSched".
+	Idle *int64 `json:"idle,omitempty" validate:"omitempty,min=0,max=1"`
```
Comment (zwzhang0107) [inline apis/slo/v1alpha1/nodeslo_types.go:35]: SchedIdle

Comment (saintube) [reply apis/slo/v1alpha1/nodeslo_types.go]: done

**Review by hormes — APPROVED** [review]

**Pull request merged**